factscore/factscorer.py

```diff
@@ -108,36 +111,39 @@ def get_score(self,
             if verbose:
                 topics = tqdm(topics)
 
-            new_topics, new_generations, new_atomic_facts = [], [], []
+            atomic_facts = []
             for topic, gen in zip(topics, generations):
                 curr_afs, _ = self.af_generator.run(gen)
                 curr_afs = [fact for _, facts in curr_afs for fact in facts]
                 if len(curr_afs)==0:
-                    # abstain from answering
+                    atomic_facts.append(None)
                     continue
-                new_topics.append(topic)
-                new_generations.append(gen)
-                new_atomic_facts.append(curr_afs)
-            topics, generations, atomic_facts = new_topics, new_generations, new_atomic_facts
-
+                atomic_facts.append(curr_afs)
+            
+            assert len(atomic_facts)==len(topics)
+            respond_ratio = np.mean([facts is not None for facts in atomic_facts])
+        
         if verbose:
             topics = tqdm(topics)
 
         scores = []
         decisions = []
         for topic, generation, facts in zip(topics, generations, atomic_facts):
+            if facts is None:
+                decisions.append(None)
+                continue
             decision = self._get_score(topic, generation, facts, knowledge_source)
             score = np.mean([d["is_supported"] for d in decision])
             decisions.append(decision)
             scores.append(score)
 
-        if return_atomic_facts:
-            return np.mean(scores), atomic_facts
-
-        if return_individual_decisions:
-            return np.mean(scores), decisions
+        if return_score_only:
+            return np.mean(scores)
```

Comment:
Do you think this is cleaner as a dictionary? or is this for backward compatibility?

Reply:
makes sense! hmm, then I wonder if just not having this option is fine since extra data doesn't hurt (and it doesn't really affect run-time).